tests/test_all.py

```diff
@@ -763,3 +768,240 @@ def test_container_suseconnect_adds_repos(container_per_test: ContainerData):
+    "named": (44, 44),
+    "app": (1654, 1654),
+    "mysql": (60, 60),
+    "dirsrv": (None, 486),
+    "stunnel": (None, 65533),
+    "polkitd": (498, 485),
+    "postgres": (None, 486),
+    "pcp": (496, 484),
+    "pcpqa": (483, 483),
+    "ldap": (498, 498),
+    "systemd-coredump": (497, 100),
+    "postfix": (51, 51),
+    "sa-milter": (497, 497),
+    "keadhcp": (None, 486),
+    "user": (1000, 1000),
+    "pulse": (498, 484),
+    "systemd-timesync": (482, 482),
+    "rtkit": (480, 480),
+}
+
+# special cases for TW & SLE 16
+if OS_VERSION in ("tumbleweed", "16.0"):
+    # these users don't use the non-default GID on TW
+    for username in (
+        "prometheus",
+        "nginx",
+        "grafana",
+        "messagebus",
+        "dirsrv",
+        "postgres",
+        "registry",
+        "keadhcp",
+    ):
+        del _USERNAME_UID_GID_MAP[username]
+
+    # completely different UID & GID on TW
+    _USERNAME_UID_GID_MAP["pcp"] = (496, 498)
+    _USERNAME_UID_GID_MAP["pcpqa"] = (494, 498)
+    _USERNAME_UID_GID_MAP["wwwrun"] = (None, 498)
+    _USERNAME_UID_GID_MAP["pesign"] = (499, 499)
+    _USERNAME_UID_GID_MAP["systemd-coredump"] = (497, 1000)
+
+
+@pytest.mark.parametrize("container", ALL_CONTAINERS, indirect=True)
+def test_uids_stable(container: ContainerData) -> None:
+    """Check that every user in :file:`/etc/passwd` has a stable uid & gid as
+    defined in ``_USERNAME_UID_GID_MAP``.
+
+    """
+    passwd: str = container.connection.file("/etc/passwd").content_string
+    assert container.connection.user("root").exists, "root user does not exist"
+
+    for userline in passwd.splitlines():
+        tmp = userline.split(":")
+        name, uid, gid = tmp[0], int(tmp[2]), int(tmp[3])
+
+        expected_uid, expected_gid = _USERNAME_UID_GID_MAP.get(
+            name, (499, 499)
+        )
+        expected_uid = 499 if expected_uid is None else expected_uid
+        expected_gid = 499 if expected_gid is None else expected_gid
+
+        # special cases
+        if (container.container.get_base().baseurl or "").split(":")[
+            0
+        ].endswith("kiosk/xorg") and name == "user":
+            expected_gid = 100
+
+        if (container.container.get_base().baseurl or "").split(":")[
+            0
+        ].endswith("kiosk/pulseaudio") and name == "messagebus":
+            expected_gid = 499
+
+        if (container.container.get_base().baseurl or "").split(":")[
+            0
+        ].endswith("kiosk/pulseaudio") and name == "polkitd":
+            expected_gid = 481
+            expected_uid = 481
+
+        if (container.container.get_base().baseurl or "").endswith(
+            "opensuse/samba-client:latest"
+        ) and name == "messagebus":
+            expected_gid = 499
+            expected_uid = 499
+
+        if (container.container.get_base().baseurl or "").endswith(
+            "opensuse/samba-server:latest"
+        ) and name == "messagebus":
+            expected_gid = 498
```

Comment:
```suggestion
            expected_uid, expected_gid = (498, 498)
```

is imho a bit better to read.